js_modules/dagster-ui/packages/ui-core/src/assets/AssetsOverview.tsx

```diff
@@ -0,0 +1,335 @@
+};
+
+const LinkToAssetGraphComputeKind = (computeKind: string) => {
+  return `/asset-groups?computeKindTags=${encodeURIComponent(JSON.stringify([computeKind]))}`;
+};
+
+const LinkToCodeLocation = (repoAddress: RepoAddress) => {
+  return `/locations/${repoAddressAsURLString(repoAddress)}/assets`;
+};
+
+export const AssetsOverview = ({viewerName}: {viewerName?: string}) => {
+  useTrackPageView();
+
+  const params = useParams();
+  const currentPath: string[] = ((params as any)['0'] || '')
+    .split('/')
+    .filter((x: string) => x)
+    .map(decodeURIComponent);
+
+  useDocumentTitle('Assets');
+
+  const assetsQuery = useQuery<AssetCatalogTableQuery, AssetCatalogTableQueryVariables>(
+    ASSET_CATALOG_TABLE_QUERY,
+    {
+      notifyOnNetworkStatusChange: true,
+    },
+  );
+  const assetCountBySection = buildAssetCountBySection(
+    assetsQuery.data?.assetsOrError.__typename === 'AssetConnection'
+      ? assetsQuery.data.assetsOrError.nodes
+      : [],
+  );
+  const {UserDisplay} = useLaunchPadHooks();
+
+  if (assetsQuery.loading) {
+    return (
+      <Page>
+        <AssetPageHeader assetKey={{path: currentPath}} />
+        <Box flex={{direction: 'row', justifyContent: 'center'}} style={{paddingTop: '100px'}}>
+          <Box flex={{direction: 'row', alignItems: 'center', gap: 16}}>
+            <Spinner purpose="body-text" />
+            <div style={{color: Colors.textLight()}}>Loading assets…</div>
+          </Box>
+        </Box>
+      </Page>
+    );
+  }
+
+  return (
+    <Box flex={{direction: 'column'}} style={{height: '100%', overflow: 'hidden'}}>
+      <AssetPageHeader
+        assetKey={{path: currentPath}}
+        right={
+          <Box flex={{gap: 12, alignItems: 'center'}}>
+            <ReloadAllButton label="Reload definitions" />
+          </Box>
+        }
+      />
+      <Box
+        padding={64}
+        flex={{justifyContent: 'center', alignItems: 'center'}}
+        style={{overflowY: 'scroll'}}
+      >
+        <Box style={{width: '60%'}} flex={{direction: 'column', gap: 16}}>
+          <Box flex={{direction: 'row', alignItems: 'center', justifyContent: 'space-between'}}>
+            <Heading>
+              {getGreeting()}
+              {viewerName ? `, ${viewerName}` : ''}
+            </Heading>
+            <Box flex={{direction: 'row', gap: 12, alignItems: 'center'}}>
+              <ViewAllAssetsLink />
+              <AssetGlobalLineageButton />
+            </Box>
+          </Box>
+          <TextInput />
+        </Box>
+      </Box>
+      <Box flex={{direction: 'column'}}>
+        {Object.keys(assetCountBySection.countsByOwner).length > 0 && (
+          <>
+            <SectionHeader sectionName="Owner" />
+            <SectionBody>
+              {Object.entries(assetCountBySection.countsByOwner).map(([label, count], idx) => (
+                <CountForAssetType key={idx} assetsCount={count}>
```

Comment:
It's good to avoid index-based keys when possible, since React will use the key to determine whether to re-render items in the list. (E.g. if a new element were added to the head of the list, *everything* would be re-keyed and therefore re-rendered. If ever item instead as a unique key, only the newly added item needs to be rendered.)

Since `label` is a key for the `countsByOwner` object, it is unique and should be used instead.

Reply:
Makes sense, thanks for the callout